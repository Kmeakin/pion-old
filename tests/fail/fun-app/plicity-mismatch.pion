//~ run = "${PION} elab ${FILE}"

let id1 = fun (A : Type) (a : A) => a;
let id2 = fun (@A : Type) (a : A) => a;

let _ : Bool = id1 @Bool true;
let _ : Bool = id2 @Bool @true;
{}
