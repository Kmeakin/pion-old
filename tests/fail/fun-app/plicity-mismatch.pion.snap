exit status: 0
stdout:
{
    let id1 : fun (A : Type) -> A -> A = fun (A : Type) (a : A) => a;
    let id2 : fun (@A : Type) -> A -> A = fun (@A : Type) (a : A) => a;
    let _ : Bool = #error;
    let _ : Bool = #error;
    ()
} : ()

stderr:
error: tried to apply implicit argument where explicit argument was expected
  ┌─ tests/fail/fun-app/plicity-mismatch.pion:7:20
  │
7 │ let _ : Bool = id1(@Bool, true);
  │                --- ^^^^^ implicit argument
  │                │    
  │                explicit function of type `fun (A : Type) -> A -> A`

error: tried to apply implicit argument where explicit argument was expected
  ┌─ tests/fail/fun-app/plicity-mismatch.pion:8:27
  │
8 │ let _ : Bool = id2(@Bool, @true);
  │                ---        ^^^^^ implicit argument
  │                │           
  │                explicit function of type `fun (@A : Type) -> A -> A`

